Add trigger on update users to remove permissions from previous orgs

lboeman opened the pull request:
Added to address #162 
Not sure if this should go somewhere else or if it can be tacked on to the framework admin sql migration.

alorenzo175 commented on datastore/migrations/0024_framework_admin.up.sql:
```diff
@@ -542,3 +542,28 @@ END;
+        INSERT INTO arbiter_data.permission_object_mapping (permission_id, object_id)
+            SELECT id, NEW.id FROM arbiter_data.permissions
+            WHERE organization_id = NEW.organization_id
+                AND object_type = 'users'
```
and ``action != 'create'``

alorenzo175 replied:
still need this

alorenzo175 commented on datastore/migrations/0024_framework_admin.up.sql:
```diff
@@ -542,3 +542,28 @@ END;
+FOR EACH ROW
+BEGIN
+    IF NEW.organization_id != OLD.organization_id THEN
+        if OLD.organization_id != get_organization_id('Unaffiliated') THEN
```
hmmm, excepting unaffiliated is to keep the default user self role/perms? Looking at that more carefully, not sure I agree that the roles and permissions should stay with unaffiliated instead of being recreated in the new org

lboeman commented:
I updated the `create_default_user_role` procedure to take an organization id as an argument and create a role there rather than unaffiliated. I had to touch 23 to do so, but I don't think there's been a tagged version since it was introduced so I think that's okay. 

Added a procedure to remove default user roles and permissions before recreation in the new org.

I've omitted an organization_id check in when deleting default roles to avoid an error condition  where an Admin user in the new organization has created a role/permission with the same name as our default roles, which would cause `add_user_to_org` to fail.

@alorenzo175 before I add tests to this would you mind taking a quick look to see how you feel about this approach?

alorenzo175 commented on datastore/migrations/0024_framework_admin.up.sql:
```diff
@@ -542,3 +542,38 @@ END;
+    -- Remove all mappings that grant actions on the user
+    DELETE FROM arbiter_data.permission_object_mapping WHERE object_id = NEW.id;
+    -- delete the default user role and specific default permissions
+    DELETE FROM arbiter_data.roles WHERE name = CONCAT('User role', BIN_TO_UUID(NEW.id, 1));
```
add a space:  'User role' => 'User role '

lboeman commented:
I think a DEFAULT prefix would be helpful for the general understanding of the user. I'll add that in there. :+1:

alorenzo175 commented on datastore/tests/test_permission_triggers.py:
```diff
@@ -146,3 +148,50 @@ def test_object_mapping_remove_obj(cursor, action, getfcn, new_permission,
+        dictcursor, new_organization, new_user):
+    user = new_user()
+    org = new_organization()
+    dictcursor.callproc(
```
add ``assert user['organization_id'] != org['id']`` just to make sure they were different from the start
